shared/src/types/hash.rs

```diff
@@ -95,6 +92,26 @@ impl TryFrom<&[u8]> for Hash {
     }
 }
 
+impl TryFrom<String> for Hash {
+    type Error = self::Error;
+
+    fn try_from(string: String) -> HashResult<Self> {
+        Ok(
+            Self(<[u8; HASH_LENGTH]>::from_hex(string).map_err(Error::FromStringError)?)
+        )
```

Comment:
here you could have used `try_into()` to avoid copy and pasting the same def

Reply:
`string.as_str().try_into()`

Reply:
true. Copy pasta game weeeeaaak